### PR TITLE
Browserify

### PR DESCRIPTION
**Pull request by zeke**
Following up on @wmhilton's work in https://github.com/npm/marky-markdown/pull/203, this PR: 

- Adds a browserify build script
- Includes a browserified file for distribution
- Uses the [browser field](https://github.com/defunctzombie/package-browser-field-spec) in package.json for easy `require`-ing
- Ensures the build is run before publishing
- Updates the "In the Browser" section of the README

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
restarting travis because i can't imagine how any of this would be breaking.

also hey! thanks for this @zeke !! will take a look when i find a sec :)

**Comment by ashleygwilliams** [comment]
would also love to hear from @wmhilton !

File: package.json [hunk 1]
```diff
@@ -3,8 +3,11 @@
   "version": "8.0.0",
   "description": "npm's markdown parser",
   "main": "index.js",
+  "browser": "dist/marky-markdown.js",
   "scripts": {
-    "test": "standard-format -w && standard && mocha --timeout 5000"
+    "build": "browserify index.js -i highlights -s markyMarkdown > dist/marky-markdown.js",
+    "test": "standard-format -w && standard && mocha --timeout 5000",
+    "prepublish": "build"
```
Comment (revin) [inline package.json]: Travis is complaining that it can't find the `build` command. `npm run build` here?

Comment (zeke) [inline package.json]: fixed!

**Comment by revin** [comment]
oops now the `dist` file isn't passing the `standard` checker _[giant overdramatic sigh]_

File: package.json [hunk 2]
```diff
@@ -3,8 +3,11 @@
   "version": "8.0.0",
   "description": "npm's markdown parser",
   "main": "index.js",
+  "browser": "dist/marky-markdown.js",
```
Comment (wmhilton) [inline package.json]: This is breaking things somehow. Investigating.

**Comment by wmhilton** [comment]
Something is very wrong. This is what I get on master:

    $ browserify -i highlights -s markyMarkdown index.js | wc -l
    46991

This is what I'm seeing in this branch

    $ browserify -i highlights -s markyMarkdown index.js | wc -l
    8

**Comment by zeke** [comment]
I see that too, @wmhilton. What the heck?

**Comment by wmhilton** [comment]
I've figured out that removing the "browser" field in the package.json magically fixes it, but I don't understand why or how to do it properly yet. Too bad I'm not a browserify expert!

**Comment by zeke** [comment]
Bizarre. I guess browserify itself is looking at the field? Chicken/egg problem?

Anyway, I've removed the `browser` key from package.json and commited a new build.

**Comment by zeke** [comment]
I forgot that for some historical reasons, the `prepublish` script is run after `npm install`, so I moved the build to `pretest` instead.

**Comment by wmhilton** [comment]
OK I think I get it now. Browserify (and now Webpack *cough* yes I might have that working too now, no promises!) prefer the raw source form rather than a precompiled form, so they can do their own minifying / deduplicating / shimming / whatever magic they do. So the "browser" field is not intended for a pre-built standalone, but as an alternative entry point. Instead of instead of having code that checks "process.browser" at runtime, you can have an alternate implementation. In fact, the browser field can be an object mapping files or dependencies to alternates or even telling browserify to ignore them.

What I've got, which seems to work, is:
```js
"browser": {
  "highlights": false
},
```

With that setting in the package.json, both browserify and webpack can "require" the module un-altered in code that is already using browserify:
```js
var mmmmmarkdown = require('marky-markdown')
```

So now the question is, do y'all still want to include a standalone compiled version in "dist"? Or is it enough that "npm install marky-markdown" work with webpack / browserify and folk can make their own standalone version if desired?

**Comment by zeke** [comment]
> do y'all still want to include a standalone compiled version

I personally wouldn't use the built version, but it's a nice thing to offer for people who are not yet drinking the browserify Kool Aid™

**Comment by wmhilton** [comment]
Alright! I give this one my seal of approval. It works with both browserify and webpack. With webpack it requires the json-loader module. <strike>I don't know how webpack handles such dependencies</strike> I looked into how and why webpack does NOT handle such dependencies and [it is a mess](https://github.com/webpack/webpack/issues/378). Short answer is webpack will inform the user that it needs a "json" loader and it is in their hands to choose an appropriate "json" loader, not the module author's. So... require('marky-markdown') with webpack will not work out-of-the-box, and that is working as intended. :confused: 

Here's my webpack command:
```shell
webpack --module-bind 'json=json' index.js bundle.js
```

I think... either just assume webpack users will figure this out, or put a note in the README: 

    Usage with webpack requires that your webpack.config.js configure a loader (such as json-loader) for .json files.

**Comment by ashleygwilliams** [comment]
i am just watching and this all just makes me very happy 💖 👻 👾 🐠 💖

**Comment by wmhilton** [comment]
Webpack apparently respects the ` "browser": {"highlights": false},` magic. I meant to say that earlier. :)

**Comment by ashleygwilliams** [comment]
hey @zeke thanks so much for this PR. is it Ready-To-Go or you still working? if you'd like a review- could you squash and update the desc to reflect that convos that happened here? thanks so much!

**Comment by zeke** [comment]
squashed and ready!

File: README.md [hunk 3]
```diff
@@ -111,12 +111,9 @@ marky-markdown some.md > some.html
 ## In the Browser
 
 This module mostly works in the browser, with the exception of the `highlights` module.
-You can compile it to a standalone file using `browserify`.
 
-```shell
-git clone https://github.com/npm/marky-markdown
-browserify marky-markdown/index.js -i highlights -s markyMarkdown > marky-markdown.js
-```
+You can `require('marky-markdown')` in scripts you browserify yourself,
```
Comment (ashleygwilliams) [inline README.md]: let's link to browserify here :)

File: README.md [hunk 4]
```diff
@@ -111,12 +111,9 @@ marky-markdown some.md > some.html
...
+or just use the standalone file in [dist/marky-markdown.js].
```
Comment (ashleygwilliams) [inline README.md]: code block please

File: README.md [hunk 5]
```diff
@@ -148,7 +149,7 @@ npm test
 - [highlights-tokens](https://github.com/zeke/highlights-tokens): A list of the language tokens used by the Atom.app [highlights](https://www.npmjs.com/package/highlights) syntax highlighter
 - [lodash](https://github.com/lodash/lodash): A utility library delivering consistency, customization, performance, &amp; extras.
 - [markdown-it](https://github.com/markdown-it/markdown-it): Markdown-it - modern pluggable markdown parser.
-- [markdown-it-emoji](https://github.com/markdown-it/markdown-it-emoji): Markdown-it-emoji extension for Markdown-it that parses markdown emoji syntax to unicode. 
+- [markdown-it-emoji](https://github.com/markdown-it/markdown-it-emoji): Markdown-it-emoji extension for Markdown-it that parses markdown emoji syntax to unicode.
```
Comment (ashleygwilliams) [inline README.md]: can we put this white space change somewhere else (new commit/PR)? thanks!

**Comment by ashleygwilliams** [comment]
ok! some very small nits- otherwise i think we are good to merge :)

**Comment by ashleygwilliams** [comment]
actually, the nits are kinda stupid. i'm just gonna merge :)

**Pull request merged**